libbeat/kibana/index_pattern_generator.go

```diff
@@ -29,33 +29,35 @@ func NewGenerator(indexName, beatName, beatDir, version string) (*IndexPatternGe
 	}
 
 	return &IndexPatternGenerator{
-		indexName:        indexName,
-		version:          version,
-		fieldsYaml:       fieldsYaml,
-		targetDirDefault: createTargetDir(beatDir, "default"),
-		targetDir5x:      createTargetDir(beatDir, "5.x"),
-		targetFilename:   beatName + ".json",
+		indexName:      indexName,
+		fieldsYaml:     fieldsYaml,
+		beatVersion:    beatVersion,
+		version:        version,
+		targetDir:      createTargetDir(beatDir, version),
+		targetFilename: beatName + ".json",
 	}, nil
 }
 
 // Create the Index-Pattern for Kibana for 5.x and default.
-func (i *IndexPatternGenerator) Generate() ([]string, error) {
+func (i *IndexPatternGenerator) Generate() (string, error) {
 	commonFields, err := common.LoadFieldsYaml(i.fieldsYaml)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	index5xPath, err := i.generate5x(commonFields)
-	if err != nil {
-		return nil, err
+	var path string
+
+	if i.version.Major == 5 {
+		path, err = i.generate5x(commonFields)
+	} else {
+		path, err = i.generate6x(commonFields)
```

Comment:
Since the version is now passed in I think we should also rename `generate5x` and `generate6x` in something more generic, like 
```
if i.version.Major >= 6 {
  path, err = i.generateMinVersion6(commonFields)
} else if i.version.Major >= 5 {
  path, err = i.generateMinVersion5(commonFields)
} else {
  return errors.New("version unsupported..")
}
```

Reply:
changed